lib/membrane_webm/parser.ex

```diff
@@ -0,0 +1,291 @@
+        :need_more_bytes
+    end
+  end
+
+  defp parse_element(bytes) do
+    case EBML.decode_element(bytes) do
+      {:ignore_element_header, element_data} ->
+        parse_many(element_data, [])
+
+      {name, type, data, rest} ->
+        element = {name, parse(data, type, name)}
+        {element, rest}
+
+      :need_more_bytes ->
+        :need_more_bytes
+    end
+  end
+
+  defp parse(bytes, :master, _name) do
+    if byte_size(bytes) == 0 do
+      []
+    else
+      parse_many(bytes, [])
+    end
+  end
+
+  defp parse(<<type::unsigned-integer-size(8)>>, :uint, :TrackType) do
+    case type do
+      1 -> :video
+      2 -> :audio
+      3 -> :complex
+      16 -> :logo
+      17 -> :subtitle
+      18 -> :buttons
+      32 -> :control
+      33 -> :metadata
+    end
+  end
+
+  defp parse(<<type::unsigned-integer-size(8)>>, :uint, :FlagInterlaced) do
+    case type do
+      # Unknown status.This value SHOULD be avoided.
+      0 -> :undetermined
+      # Interlaced frames.
+      1 -> :interlaced
+      # No interlacing.
+      2 -> :progressive
+    end
+  end
+
+  defp parse(<<type::unsigned-integer-size(8)>>, :uint, :ChromaSitingHorz) do
+    case type do
+      0 -> :unspecified
+      1 -> :left_collocated
+      2 -> :half
+    end
+  end
+
+  defp parse(<<type::unsigned-integer-size(8)>>, :uint, :ChromaSitingVert) do
+    case type do
+      0 -> :unspecified
+      1 -> :top_collocated
+      2 -> :half
+    end
+  end
+
+  # per RFC https://datatracker.ietf.org/doc/html/rfc8794#section-7.1
+  defp parse(<<>>, :integer, _name) do
+    0
+  end
+
+  defp parse(bytes, :integer, _name) do
+    s = bit_size(bytes)
+    <<num::signed-big-integer-size(s)>> = bytes
+    num
+  end
+
+  # per RFC https://datatracker.ietf.org/doc/html/rfc8794#section-7.2
+  defp parse(<<>>, :uint, _name) do
+    0
+  end
+
+  defp parse(bytes, :uint, _name) do
+    :binary.decode_unsigned(bytes, :big)
+  end
+
+  # per RFC https://datatracker.ietf.org/doc/html/rfc8794#section-7.3
+  defp parse(<<>>, :float, _name) do
+    0
+  end
+
+  defp parse(<<num::float-big>>, :float, _name) do
+    num
+  end
+
+  # The demuxer MUST only open webm DocType files.
+  # per demuxer guidelines https://www.webmproject.org/docs/container/
+  defp parse(bytes, :string, :DocType) do
+    case parse(bytes, :string, nil) do
+      "webm" -> "webm"
+      type -> raise "The file DocType is '#{type}' but it MUST be 'webm'"
+    end
+  end
+
+  defp parse(bytes, :string, :CodecID) do
+    case parse(bytes, :string, nil) do
+      "A_OPUS" -> :opus
+      "A_VORBIS" -> :vorbis
+      "V_VP8" -> :vp8
+      "V_VP9" -> :vp9
+    end
+  end
+
+  defp parse(bytes, :string, _name) do
+    chars = for <<c::utf8 <- bytes>>, do: <<c::utf8>>
+    chars |> Enum.take_while(fn c -> c != <<0>> end) |> Enum.join()
+  end
+
+  defp parse(bytes, :utf_8, _name) do
+    bytes
+    |> String.codepoints()
+    |> Enum.reduce("", fn codepoint, result ->
+      <<parsed::8>> = codepoint
```

Comment:
As utf-8 can take more than 1 byte for a codepoint this can result in pattern match error.